pkg/providers/docker/reconciler/reconciler_test.go

```diff
@@ -103,7 +167,7 @@ type reconcilerTest struct {
 	datacenterConfig     *anywherev1.DockerDatacenterConfig
 }
 
-func newReconcilerTest(t testing.TB) *reconcilerTest {
+func newReconcilerTest(t testing.TB, opts ...clusterOpt) *reconcilerTest {
```

Comment:
I would vote against this
I get that it makes the setup code flow a bit better, but it only works for cluster modification, and it opens the door to want to do the same with other objects, which is not possible because go only accepts one set of variadic arguments. In that case, we would need two different patterns to update two different objects and that, imo, makes the tests more confusing.

Reply:
Noted, that makes sense. I've updated it taken the feedback into consideration.